BitmarkRegistry/Extensions/Array+Transaction+Extension.swift

```diff
@@ -0,0 +1,26 @@
+//
+//  Array+Transaction+Extension.swift
+//  BitmarkRegistry
+//
+//  Created by Macintosh on 6/30/19.
+//  Copyright © 2019 thuyentruong. All rights reserved.
+//
+
+import Foundation
+import BitmarkSDK
+
+extension Array where Iterator.Element == Transaction {
+  func firstIndexWithId(_ txId: String) -> Int? {
+    return self.firstIndex(where: { $0.id == txId })
+  }
+
+  /** - Returns: the unique asc-offset bitmarks */
+  mutating func removeObsoleteTxs() {
+    let descTxs = self.sorted(by: { $0.offset > $1.offset })
+    self = descTxs.reduce(into: [Transaction](), { (txs, tx) in
```

Comment:
Can we apply `filter`?
And it's better doing `filter` first, then `sort`.

Reply:
the request returns the transactions is in asc-offset order.

But removeObsolete, I want to keep the transaction which has the highest offset and remove all transactions with same id but lower offset (which are old)

That's why I don't think we can apply ```filter``` first, cause it will start with lower offset? 🤔

Reply:
Ok 👍